scripts/marimo_example.py

```diff
@@ -7,89 +7,90 @@
+
+        if opts is not None:
+            drawer.SetDrawOptions(opts)
+
+        drawer.DrawMolecule(mol)
+        drawer.FinishDrawing()
+        return mo.Html(drawer.GetDrawingText())
+
     solubility_range = mo.ui.range_slider(
         -10, 2, 0.5,
         debounce=True,show_value=True,full_width=True,
         label="Solubility",
     )
-
-    molwt_range = mo.ui.range_slider(
-        0, 600, 10,
-        debounce=True, show_value=True, full_width=True,
-        label="MolWt",
-    )
-
-    # logp_range = mo.ui.range_slider(
-    #     -10, 10, 1,
-    #     debounce=True, show_value=True, full_width=True,
-    #     label="LogP",
-    # )
-
-    # num_h_donors_range = mo.ui.range_slider(
-    #     0, 20, 1,
-    #     debounce=True, show_value=True, full_width=True,
-    #     label="NumHDonors",
-    # )
-
-    # num_h_acceptors_range = mo.ui.range_slider(
-    #     0, 20, 1,
-    #     debounce=True, show_value=True, full_width=True,
-    #     label="NumHAcceptors",
-    # )
-    return molwt_range, solubility_range
+    return mol_to_svg, solubility_range
 
 
 @app.cell
-def load_data_and_grid(Descriptors, datafiles, mols2grid):
+def create_grid(SOLUBILITY_SDF, mols2grid):
     # NOTE:
     # This cell is intentionally kept independent from the sliders.
     # In marimo, cells are re-executed whenever any of their dependencies change.
     # Keeping grid creation here prevents MolGrid.from_sdf(...) from being
     # re-run on every slider update, which would reset the widget state.
-
-    grid = mols2grid.MolGrid.from_sdf(datafiles.SOLUBILITY_SDF, size=(120, 100))
-    df = grid.dataframe
-    df["MolWt"] = df["mol"].apply(Descriptors.ExactMolWt)
-    # df["LogP"] = df["mol"].apply(Descriptors.MolLogP)
-    # df["NumHDonors"] = df["mol"].apply(Descriptors.NumHDonors)
-    # df["NumHAcceptors"] = df["mol"].apply(Descriptors.NumHAcceptors)
-    view = grid.display(n_items_per_page=12)
-    return df, grid, view
+    grid = mols2grid.MolGrid.from_sdf(SOLUBILITY_SDF, size=(120, 100))
+    sel_ids = grid.get_selection_state()
+    view = grid.display(n_items_per_page=12, selection=True)
+    return grid, sel_ids, view
 
 
 @app.cell
-def apply_filters_and_layout(df, grid, mo, molwt_range, solubility_range, view):
-    filters = {
-        "SOL": solubility_range.value,
-        "MolWt": molwt_range.value,
-        # "LogP": logp_range.value,
-        # "NumHDonors": num_h_donors_range.value,
-        # "NumHAcceptors": num_h_acceptors_range.value,
-    }
-
-    mask = True
-    for col, (lo, hi) in filters.items():
-        mask &= df[col].between(lo, hi)
-
-    results = df.loc[mask]
+def filter_and_display(grid, mo, solubility_range, view):
+    mask = grid.dataframe["SOL"].between(*solubility_range.value)
+    results = grid.dataframe.loc[mask]
+    # Same as:
+    # results = grid.dataframe.loc[
+    #     (grid.dataframe["SOL"] >= solubility_range.value[0]) & (grid.dataframe["SOL"] <= solubility_range.value[1])
+    # ]
```

Comment:
should be moved one line up as:
```suggestion
    # Same as:
    # grid.dataframe["SOL"] >= solubility_range.value[0]) & (grid.dataframe["SOL"] <= solubility_range.value[1])
```